app/scripts/directives/processTemplateDialog.js

```diff
@@ -4,20 +4,49 @@
   angular.module('openshiftConsole').component('processTemplateDialog', {
     controller: [
       '$scope',
+      '$filter',
+      '$timeout',
+      'Catalog',
       'DataService',
+      'KeywordService',
+      'NotificationsService',
+      'ProjectsService',
+      'RecentlyViewedProjectsService',
       ProcessTemplateDialog
     ],
     controllerAs: '$ctrl',
     bindings: {
       template: '<',
+      project: '<',
+      useProjectTemplate: '<',
       onDialogClosed: '&'
     },
     templateUrl: 'views/directives/process-template-dialog.html'
   });
 
-  function ProcessTemplateDialog($scope, DataService) {
+  function ProcessTemplateDialog($scope,
+                                 $filter,
+                                 $timeout,
+                                 Catalog,
+                                 DataService,
+                                 KeywordService,
+                                 NotificationsService,
+                                 ProjectsService,
+                                 RecentlyViewedProjectsService) {
     var ctrl = this;
     var validityWatcher;
+    var projectListWatcher;
+    var MAX_PROJECTS_TO_WATCH = 250;
```

Comment:
Interested in this constant, wondering if it should be configurable via constants.js?

Curious if we will actually be watching 250 projects at one time.  Thats a lot of watches 😄

Reply:
I followed the lead from https://github.com/openshift/origin-web-console/blob/master/app/scripts/controllers/projects.js#L22

We should probably define a constant somewhere. I will add a comment for now.